### PR TITLE
Document IDP PR / Release scripts

### PR DESCRIPTION
**Pull request by matthinz**
This PR updates the deployment guide to include information about the `create-deploy-pr` and `create-release` scripts added in 18f/identity-idp#9859 and 18f/identity-idp#9929.

Preview: https://federalist-7d0b2b76-42fc-4df1-9825-8c3cd77e0a15.sites.pages.cloud.gov/preview/18f/identity-handbook/matthinz/deploy-automation-lite/articles/appdev-deploy.html

### CODE REVIEW

**Review by zachmargolis — APPROVED** [review]

File: _articles/appdev-deploy.md [hunk 1]
```diff
@@ -217,8 +229,13 @@ Staging used to be deployed by this process, but this was changed to deploy the
 
 ##### IDP
 
+The IDP includes a script to create a release based on a merged pull request.  It relies on [`gh`](https://cli.github.com/), the Github cli. Install that first (`brew install gh`) and get it connected to the identity-idp repo. Then, run the script to create a release:
+
+```shell
+scripts/create-relase <PR_NUMBER>
```
Comment (zachmargolis) [inline _articles/appdev-deploy.md:235]:
```suggestion
scripts/create-release <PR_NUMBER>
```

File: _articles/appdev-deploy.md [hunk 2]
```diff
@@ -78,16 +78,32 @@ Once you've run through proofing in staging, the next step is to cut a release f
...
+##### IDP
 
-For IdP:
-```bash
-cd identity-idp
-git fetch
-git checkout $(curl --silent https://idp.staging.login.gov/api/deploy.json | jq -r .git_sha)
-git checkout -b stages/rc-2024-01-09 # CHANGE THIS DATE
-git push -u origin HEAD
+The IDP includes a script to create deployment PRs. It relies on [`gh`](https://cli.github.com/), the Github cli. Install that first (`brew install gh`) and get it connected to the identity-idp repo. Then, run the script to create a PR:
```
Comment (aduth) [inline _articles/appdev-deploy.md:83]: I'm running thru this for the first time and am very unclear what "get it connected" means at a glance. Will see if I can figure it out, but figured I'd share my first impression reading this.

Comment (zachmargolis) [reply _articles/appdev-deploy.md:83]: `gh login` I think

Comment (aduth) [reply _articles/appdev-deploy.md:83]: Yeah, my experience wasn't so great.

1. Installed `gh` via documented `brew install gh`
2. Don't know what's meant by "get it connected" 🤷 
3. Proceed to `scripts/create-deploy-pr`
4. Command half-completes with error after "Creating PR...": "To get started with GitHub CLI, please run:  gh auth login"
5. Am now in a fragmented state where a branch has already been pushed to IdP for `stages/rc-2024-01-29` but there's no pull request
6. I try `gh auth login` and go through a series of prompts
7. It gets stuck at "Press Enter to open github.com in your browser...", pressing <kbd>Enter</kbd> doesn't do anything. Apparently this is a [known issue](https://github.com/cli/cli/issues/2422)?
8. Try going directly to https://github.com/login/device via the comments
9. See sketchy prompt asking for a ton of different permissions to my GitHub account from a location I'm not familiar with. Nope'd the heck out of there.
10. Go back to the manual process, which I'd probably have to do in the fragmented state anyways

**Pull request merged**